### DISCUSSION TITLE
AIR security

### DISCUSSION THREAD

**Discussion by dgurney75** [airsdk/Adobe-Runtime-Support#738] [Q&A]
I am being asked by my employer if AIR for mobile app development poses a security risk by using it since it utilizes Flash technology. Is there any white paper or other references I can send to them to ease their concerns. I love developing with AIR and Animate CC and would hate to see it halted because of this. BTW, an anonymous dev sent upper management this link which caused the concerns. I looked into it and most of the security issues seem to be desktop related and the last SDK they reference is AIR 16. 

https://www.cvedetails.com/vulnerability-list/vendor_id-53/product_id-11602/Adobe-Adobe-Air.html

Any help would be greatly appreciated to help my case to continue to develop with AIR.

**Comment by ajwfrost** [comment]
Hi
In the past, Adobe AIR included Adobe Flash Player, which is why it got listed in all those CVE issues.. Adobe decided to stop shipping the Flash Player with it quite a few years ago. So, it uses the same technology as the Flash Player did, but the attack vector is somewhat different...

In order to get attacked if you were using the Flash Player in a browser, you just had to be browsing a page where a third party advert could be added in which someone had placed a SWF that then automatically caused the Flash Player to load this and execute the SWF...
If you're using Adobe AIR then you're creating the SWF file and specifying that the runtime should load it. As long as you haven't coded any exploits into this, you should be okay..!

The one part I would be careful of is any web-based interactions so e.g. loading a page into the HTMLLoader class where you are then opening up a web page if any part of this isn't under your control. Internally, AIR uses a very old version of WebKit for this, so this is one part I would suggest avoiding.. but you mention that this is for a mobile app? in which case, no WebKit is available and it's just the raw AIR runtime that's available.

Other than that it seems to me that AIR is about as secure as Word or Excel or any other desktop-based software component. If you're able to run an AIR application, it implies you're sat at a computer and already have access to it. To open yourself up to a remote agent getting access to your computer, you'd need to do something within your AIR app to grant them that access: AIR of itself won't do anything like that and the triggers would all come from your app. 

In case it helps, we would fix any vulnerabilities we're made aware of. Adobe said the same thing when they did their end of life announcement: 
https://blog.adobe.com/en/publish/2019/05/30/the-future-of-adobe-air.html
> Adobe will provide basic security support – limited to security fixes only for desktop platforms (Windows 7 and above, and Mac OS X) – for Adobe AIR v32 until the end of 2020

... and of course, there were zero security fixes needed in all that time.

So as long as you don't do anything silly in AIR, you should be fine, and of course it's equally possible to be silly in any other framework or native Android/iOS application...

Hope that helps!

**Reply by dgurney75** [reply]
One thing i know would help with this is to update the SSL, TLS, and Cipher suites to use the latest versions. Also AIR should be using at least SHA-256 for encryption. And what should we use for Android for encryption since ELS isn't supported. Let me know what you think.
